lib/ua_update_messaging.py

```diff
@@ -0,0 +1,167 @@
+    """Return human-readable message if ESM is offered on this LTS release.
+
+    :param cfg: UAConfig instance for this environment.
+    :return: string if contract expiry notice is needed.
+    """
+    series = util.get_platform_info()["series"]
+
+    if series == "trusty":
+        return None  # Trusty has no esm-apps offering
+
+    out, _err = util.subp(["/usr/bin/ubuntu-distro-info", "--supported-esm"])
+    if series in out:
+        out, _err = util.subp(
+            ["/usr/bin/ubuntu-distro-info", "--series", series, "-yeol"]
+        )
+        is_active_esm = bool(int(out) <= 0)
+
+        config_allow_beta = util.is_config_value_true(
+            config=cfg.cfg, path_to_value="features.allow_beta"
+        )
+        ent_cls = entitlements.ENTITLEMENT_CLASS_BY_NAME["esm-apps"]
+        if config_allow_beta or (not ent_cls.is_beta and is_active_esm):
+            return status.MESSAGE_ANNOUNCE_ESM
+    return None
+
+
+def update_apt_and_motd_messages(
+    cfg: config.UAConfig, msg_dir: str = defaults.DEFAULT_MESSAGING_DIR
+) -> None:
+    """Emit human-readable status messages in msg_dir.
+
+    These structured messages will be sourced by both /etc/update.motd.d
+    and APT UA-configured hooks. APT hook content will orginate from
+    apt-hook/hook.cc.
+
+    :param cfg: UAConfig instance for this environment.
+    :param msg_dir: The directory in which to persist message files.
+    """
+    setup_logging(logging.INFO, logging.DEBUG)
+    logging.debug("Updating UA messages for APT and MOTD.")
+    if not os.path.exists(msg_dir):
+        os.makedirs(msg_dir)
+
+    motd_expiry_msg, apt_expiry_msg = format_contract_expiry_messages(cfg)
+    motd_expiry_file = os.path.join(
+        msg_dir, ExternalMessage.CONTRACT_EXPIRY_MOTD.value
+    )
+    if motd_expiry_msg:
+        util.write_file(motd_expiry_file, motd_expiry_msg)
+    else:
+        # Clean template and rendered output file
+        util.remove_file(motd_expiry_file)
+        util.remove_file(motd_expiry_file.replace(".tmpl", ""))
+
+    apt_expiry_file = os.path.join(
+        msg_dir, ExternalMessage.CONTRACT_EXPIRY_APT.value
+    )
+    if apt_expiry_msg:
+        util.write_file(apt_expiry_file, apt_expiry_msg)
+    else:
+        # Clean template and rendered output file
+        util.remove_file(apt_expiry_file)
+        util.remove_file(apt_expiry_file.replace(".tmpl", ""))
+    # Refresh messaging templates with apt pkg counts for any updated messaging
+    util.subp(["/usr/lib/ubuntu-advantage/apt-esm-hook", "process-templates"])
+
+    # Announce ESM availabilty on LTS releases
+    esm_news_msg = format_esm_announcement_message(cfg)
+    esm_news_file = os.path.join(msg_dir, ExternalMessage.ESM_ANNOUNCE.value)
+    if esm_news_msg:
+        util.write_file(esm_news_file, esm_news_msg)
+    else:
+        util.remove_file(esm_news_file)
+        util.remove_file(esm_news_file.replace(".tmpl", ""))
```

Comment:
Currently esm_news_file isn't a `tmpl` file so we dont need this line
```suggestion
    else:
        util.remove_file(esm_news_file)
```